packages/commons/src/events/state/flags.ts

```diff
@@ -127,11 +131,26 @@ export function resolveEventCustomFlags(
   event: EventDocument,
   eventConfiguration: EventConfig
 ): CustomFlag[] {
-  const acceptedActions = getAcceptedActions(event)
-  const actions = acceptedActions
+  const sortedActions = getAcceptedActions(event)
     .filter(({ type }) => !isMetaAction(type))
     .sort((a, b) => a.createdAt.localeCompare(b.createdAt))
 
+  // First find indices of all declare actions
+  const declareIndexes = sortedActions
+    .map((a, i) => (a.type === ActionType.DECLARE ? i : -1))
+    .filter((i) => i !== -1)
+
+  let actions = sortedActions
+
+  // If there is more than one declare action, lets filter out all actions between the second last and last declare actions
+  // Why is this done? This is to handle 'redeclaration' cases, i.e. when a user edits a declared record and then does 'Declare with edits'
+  // Then we want to ignore all the actions between the second last and last declare actions.
+  if (declareIndexes.length >= 2) {
+    const secondLast = declareIndexes[declareIndexes.length - 2]
+    const last = declareIndexes[declareIndexes.length - 1]
+    actions = sortedActions.filter((_, idx) => idx < secondLast || idx >= last)
```

Comment:
```suggestion
    actions = [
       ...sortedActions.slice(0, secondLast),
       ...sortedActions.slice(last)
     ]
```
might be a tad more readable